js/app/test/cancel_events.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { test, expect } from "@gradio/tootils";
+
+test("yielding", async ({ page }) => {
+	let first_iteration;
+	let fourth_iteration;
+	let last_iteration;
+
+	page.on("websocket", (ws) => {
```

Comment:
We listen for the websocket event, so we know it exists and can start listeneing for specific messages as soon as possible, we don't want the websocket connection to progress before we had a change to listen to it.